CMakeLists.txt

```diff
@@ -1,9 +1,25 @@
 # Copyright (c) 2021 Nordic Semiconductor ASA
+# Copyright (c) 2022 Legrand North America, LLC.
 # SPDX-License-Identifier: Apache-2.0
 #
 # This CMake file is picked by the Zephyr build system because it is defined
 # as the module CMake entry point (see zephyr/module.yml).
 
+
+# Verify the module name is set correctly
+set(expected_module_name "example-application")
+if(NOT "${module_name}" STREQUAL ${expected_module_name})
+  message(FATAL_ERROR "
+    ------------------------------------------------------------------------
+    A module configuration error has occurred.
+    
+    This module is named '${module_name}' when '${expected_module_name}' is expected. 
+    Verify `module.yml` contains the line 'name: ${expected_module_name}'.
+    ------------------------------------------------------------------------")
+endif()
+unset(expected_module_name)
+
+
```

Comment:
we should definitely not have this in every application, if we do and it is required, then Zephyr needs to solve the problem, not apps.

Reply:
This is here simply because Zephyr hasn't **required** the module name to be set. 

As long as the module name is implied from the directory name then it can be changed by an importing manifest, and any usage of `${ZEPHYR_<moduleNameUpper>_MODULE_DIR} is suspect.  This is here to alert **integrators** when a module name has been inadvertently changed.

Reply:
I'm sorry, but what is an integrator in this context?

Reply:
It is the Firmware engineer that sets the configuration and unique data values (e.g., model number) for building a binary image. It is a role I have fulfilled in a prior job. The integration may not involve any product-unique logic.

In Zephyr, the integrator is the person that fills in the prj.conf, CMakeLists.txt,  twister-recognizable .yaml file, README.rst, and (possibly empty) source *.c file(s).

In ISO/IEC 26580:2021 terminology, this is Application Engineering as opposed to Domain Engineering.  The latter is responsible for developing reusable processes and assets (e.g., drivers, subsystems, board configurations, overlays, etc. along with the requirements, verification artifacts, and documentation).

Reply:
No, we should not do this in Zephyr modules.